core-ui/src/state/authDataAtom.ts

```diff
@@ -0,0 +1,14 @@
+import { atom, RecoilState } from 'recoil';
+
+type AuthDataState = {
+  'client-certificate-data': string;
+  'client-key-data': string;
+  token?: string;
+} | null;
```

Comment:
It's "either-or" here: we should have either:
- `null` or
- {token: string} or
- {'client-certificate-data': string, 'client-key-data': string}